### PR TITLE
ethcore/res: remove morden testnet

### PR DESCRIPTION
**Pull request by q9f**
ethereum foundation no longer uses the morden testnet. it was replaced by ropsten (ethash).
- https://github.com/ethereum/ropsten

ethereum classic no longer uses the morden testnet. it was replaced by mordor (ethash) and kotti (clique).
- https://github.com/eth-classic/mordor
- https://github.com/goerli/testnet#meta-data-kotti-classic

the last two clients supporting morden are parity ethereum (this) and geth classic. geth classic reached end of life on january 15, 2020 leaving parity with the only client supporting the deprecated testnet.
- https://github.com/etclabscore/go-ethereum/releases

this pull request removes morden testnet.

this pull request replaces morden in tests with either null_morden or ropsten whereever applicable. 

- [x] You added a brief description of the PR, e.g.:
  - What does it do?
  - What important points reviewers should know?
  - Is there something left for follow-up PRs?
- [ ] You labeled the PR with appropriate labels if you have permissions to do so.
- [ ] You mentioned a related issue if this PR related to it, e.g. `Fixes #228` or `Related #1337`.
- [x] You asked any particular reviewers to review. If you aren't sure, start with GH suggestions.
- [x] Your PR adheres [the style guide](https://wiki.parity.io/Coding-guide)
  - In particular, mind the maximal line length.
  - There is no commented code checked in unless necessary.
  - Any panickers have a proof or removed.
- [ ] You updated any rustdocs which may have changed

### CODE REVIEW

**Comment by dvdplm** [comment]
Thanks for the PR!
The two test failures (`on_close_block_with_uncle` and `on_close_block`) seem legit; can you investigate what might be wrong there?

**Comment by q9f** [comment]
using ropsten instead of morden in tests broke some assumptions. I updated the tests but their logic should be reviewed carefully.

File: ethcore/src/tests/client.rs [hunk 1]
```diff
@@ -80,7 +80,7 @@ fn should_return_registrar() {
 		IoChannel::disconnected(),
 	).unwrap();
 	let address = client.registrar_address();
-	assert_eq!(address, Some("52dff57a8a1532e6afb3dc07e2af58bb9eb05b3d".parse().unwrap()));
+	assert_eq!(address, Some("0x81a4b044831c4f12ba601adb9274516939e9b8a2".parse().unwrap()));
```
Comment (ordian) [inline ethcore/src/tests/client.rs]:
```
---- tests::client::should_return_registrar stdout ----
thread 'tests::client::should_return_registrar' panicked at 'called `Result::unwrap()` on an `Err` value: Invalid character 'x' at position 1', src/libcore/result.rs:1165:5
```
```suggestion
	assert_eq!(address, Some("81a4b044831c4f12ba601adb9274516939e9b8a2".parse().unwrap()));
```

**Comment by ordian** [comment]
try this
```patch
diff --git a/ethcore/spec/src/chain.rs b/ethcore/spec/src/chain.rs
index 1112fb082..f714530f0 100644
--- a/ethcore/spec/src/chain.rs
+++ b/ethcore/spec/src/chain.rs
@@ -146,8 +146,8 @@ mod tests {
                assert_eq!(s.balance(&"0000000000000000000000000000000000000002".parse().unwrap()).unwrap(), 1u64.into());
                assert_eq!(s.balance(&"0000000000000000000000000000000000000003".parse().unwrap()).unwrap(), 1u64.into());
                assert_eq!(s.balance(&"0000000000000000000000000000000000000004".parse().unwrap()).unwrap(), 1u64.into());
-               assert_eq!(s.balance(&"102e61f5d8f9bc71d0ad4a084df4e65e05ce0e1c".parse().unwrap()).unwrap(), U256::from(1u64) << 200);
-               assert_eq!(s.balance(&"0000000000000000000000000000000000000000".parse().unwrap()).unwrap(), 0u64.into());
+               assert_eq!(s.balance(&"874b54a8bd152966d63f706bae1ffeb0411921e5".parse().unwrap()).unwrap(), U256::from(1000000000000000000000000000000u128));
+               assert_eq!(s.balance(&"0000000000000000000000000000000000000000".parse().unwrap()).unwrap(), 1u64.into());
        }
 
        #[test]
@@ -155,9 +155,9 @@ mod tests {
                let tempdir = TempDir::new("").unwrap();
                let ropsten = new_ropsten(&tempdir.path());
 
-               assert_eq!(ropsten.state_root, "f3f4696bbf3b3b07775128eb7a3763279a394e382130f27c21e70233e04946a9".parse().unwrap());
+               assert_eq!(ropsten.state_root, "217b0bbcfb72e2d57e28f33cb361b9983513177755dc3f33ce3e7022ed62b77b".parse().unwrap());
                let genesis = ropsten.genesis_block();
-               assert_eq!(view!(BlockView, &genesis).header_view().hash(), "0cd786a2425d16f152c658316c423e6ce1181e15c3295826d7c9904cba9ce303".parse().unwrap());
+               assert_eq!(view!(BlockView, &genesis).header_view().hash(), "41941023680923e0fe4d74a34bdac8141f2540e3ae90623718e47d66d1ca4a2d".parse().unwrap());
        }
 
        #[test]
```

**Comment by q9f** [comment]
tests pass. might need release notes if someone still uses morden.

**Pull request merged**